tony-history-server/app/utils/SecurityUtils.java

```diff
@@ -0,0 +1,82 @@
+package utils;
+
+import com.typesafe.config.Config;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.HdfsConfiguration;
+import org.apache.hadoop.security.UserGroupInformation;
+
+import java.io.File;
+import java.io.IOException;
+import org.apache.hadoop.yarn.api.ApplicationConstants;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+import play.Logger;
+
+/**
+ * The class handles authentication when cluster is security enabled
+ */
+public class SecurityUtils {
+  private static final Logger.ALogger LOG = Logger.of(SecurityUtils.class);
+  private static Config appConf;
+  private static FileSystem fs;
+
+  private static final String HADOOP_CONF_DIR = ApplicationConstants.Environment.HADOOP_CONF_DIR.key();
+  private static final String CORE_SITE_CONF = YarnConfiguration.CORE_SITE_CONFIGURATION_FILE;
+  private static final String HDFS_SITE_CONF = "hdfs-site.xml";
+
+  private static SecurityUtils instance = null;
+
+  public static SecurityUtils getInstance(Config conf) {
+    if (instance == null) {
+      instance = new SecurityUtils(conf);
+    }
+
+    return instance;
+  }
+
+  private SecurityUtils(Config appConf) {
```

Comment:
This looks weird. Normally Utils should only have static methods.

Reply:
I was inspired by dr-elephant code. https://github.com/linkedin/dr-elephant/blob/master/app/com/linkedin/drelephant/security/HadoopSecurity.java

I guess the point is to have only instance of SecurityUtil for all controllers since they all use the same privilege in kerberos, so if it's already instantiated from one of the controllers, no need to make a new one, but simple just return the SecurityUtil instance from `getInstance()`.

Reply:
ok.. rename to something other than Utils :) Utils are almost always a static class with private constructors.

Reply:
I think SecurityUtils can be all static methods. The methods should take in a `Configuration` object if needed.

Reply:
Ok. Fixed: https://github.com/linkedin/TonY/pull/62/commits/fb5d61579eea00e50d6ab9ae6993d69a3f89eaba

Reply:
I don't think there should be a getInstance() method in SecurityUtils if it's all static methods. Like Keqiu said, generally Utils classes are singletons and there's no need to get an instance of them; you just call the static methods directly.

Reply:
Hmm no that I think about it, this module shouldn't be in utils because it's not really a util: it's a class that only needed to be created once and share by all clients (controllers). Thus, I'm moving it to its own folder called `security`, and refactored everything accordingly. Commit: https://github.com/linkedin/TonY/pull/62/commits/89bd2fdd13e2bc207eb4e0a28e4025555cdeb514. Let me know what you think @erwa @oliverhu